source/common/common/logger.cc

```diff
@@ -26,14 +36,22 @@ std::vector<Logger>& Registry::allLoggers() {
   return *all_loggers;
 }
 
-void LockingStderrSink::log(const spdlog::details::log_msg& msg) {
-  Thread::OptionalLockGuard<Thread::BasicLockable> guard(lock_);
-  std::cerr << msg.formatted.str();
+void LockingStderrOrFileSink::log(const spdlog::details::log_msg& msg) {
+  if (log_file_) {
+    log_file_->write(msg.formatted.str());
+  } else {
+    Thread::OptionalLockGuard<Thread::BasicLockable> guard(lock_);
+    std::cerr << msg.formatted.str();
```

Comment:
Can you add a comment on how it's serializing inside the file in the `log_file_` case, hence why we don't nee to lock as in the `stderr` case?